tenpy/networks/mps.py

```diff
@@ -5289,6 +5306,26 @@ def _contract_with_RP(self, C, i):
         return C
 
 
+class MPSEnvironmentJW(MPSEnvironment):
+    """Class similar to :class:`MPSEnvironment`, but overwriting the method
+    :meth:`_contract_LP` in a way that the JW string is automatically added when
+    contracting the left environment only -> expectation values should only
+    be calculated from left to right with this class. This works for finite MPS."""
```

Comment:
Done, however, I noticed that `MPSEnvironment` does not have an attribute `bc` -- instead there's `ket.bc` and `bra.bc`. I'm not sure if there's a special case, where `bra.bc`  !=  `ket.bc`, otherwise does it make sense to set a `bc` attribute for the `MPSEnvironment` (or respectively setting this in `BaseEnvironment`)  ?